src/System/extract_state.rs

```diff
@@ -0,0 +1,129 @@
+use std::collections::HashMap;
+use std::{error::Error, fmt};
+
+use edbm::zones::OwnedFederation;
+
+use crate::component::Declarations;
+use crate::extract_system_rep::SystemRecipe;
+use crate::EdgeEval::constraint_applyer::apply_constraints_to_state;
+use crate::ModelObjects::component::State;
+use crate::ModelObjects::representations::QueryExpression;
+use crate::TransitionSystems::{LocationID, LocationTuple, TransitionSystemPtr};
+
+/// This function takes a QueryExpresssion, the system recipe, and the transitionsystem -
+/// to define a state from the QueryExpression which has clocks and locations.
+/// The QueryExpression looks like this: State(Vec<LocName>, Option<BoolExpression>)
```

Comment:
Describe parameters

Reply:
Done

Reply:
Can you use markdown-formatting and referencing (``[`QueryExpression\`]``, `` `state_query` ``, etc.)`

Reply:
Solved @t-lohse